### PR TITLE
Page Pattern: Render patterns on client side

### PR DESCRIPTION
**Pull request by arthur791004**
#### Proposed Changes

* Render page patterns on the ETK plugin and get rid of the preview endpoint and mshot

See p1666257227083559-slack-CRWCHQGUB for discussions.

https://user-images.githubusercontent.com/13596067/199210334-b31e66e2-f737-423a-a335-ca9115147317.mov

#### Testing Instructions

* Sandbox both the API and your site 
* Run `cd apps/editing-toolkit && yarn dev --sync`. Or run `install-plugin.sh editing-toolkit feat/render-page-patterns-on-client-side` on your sandbox.
* Go to your site and create a new page
* Verify the site title, tagline and logo are rendered correctly

#### Pre-merge Checklist

- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to https://github.com/Automattic/wp-calypso/issues/69555

### CODE REVIEW

File: packages/page-pattern-modal/src/components/page-pattern-modal.tsx [hunk 1]
```diff
@@ -46,32 +44,46 @@ class PagePatternModal extends Component< PagePatternModalProps, PagePatternModa
...
-				? parseBlocks( replacePlaceholders( html, this.props.siteInformation ) )
-				: [];
-			return prev;
-		}, {} as Record< string, BlockInstance[] > );
+	getFormattedPatternsByPatternSlugs = memoize( ( patterns: PatternDefinition[] ) => {
+		const blocksByPatternSlugs = patterns.reduce(
+			( prev, { name, description = '', html, pattern_meta } ) => {
+				const viewportWidth = pattern_meta?.viewport_width
+					? Number( pattern_meta.viewport_width )
+					: 1280;
```
Comment (miksansegundo) [inline packages/page-pattern-modal/src/components/page-pattern-modal.tsx:53]: Why `1280`? Maybe we can add a comment.

File: packages/page-pattern-modal/src/components/page-pattern-modal.tsx [hunk 2]
```diff
@@ -46,32 +44,46 @@ class PagePatternModal extends Component< PagePatternModalProps, PagePatternModa
...
+
+				prev[ name ] = {
+					name,
+					// Keep showing the description as before instead of the title
+					title: description,
+					description: '',
+					blocks: html
+						? parseBlocks( replacePlaceholders( html, this.props.siteInformation ) )
+						: [],
+					viewportWidth: Math.max( viewportWidth, 320 ),
```
Comment (miksansegundo) [inline packages/page-pattern-modal/src/components/page-pattern-modal.tsx:62]: Why `320`?

**Review by miksansegundo — COMMENTED** [review]
I found an issue with the height of the page pattern. It seems related to the viewport height.

The pattern height looks off: 

Screenshot 2565-12-15 at 16 57 59

It looks good if the patterns load when the viewport has a shorter height: 

Screenshot 2565-12-15 at 17 00 00

I can only reproduce it by opening the Dev Tools, which reduces the viewport height. It doesn't get fixed when I resize the window 🤔. 

https://user-images.githubusercontent.com/1881481/207831481-af050968-f55a-41e3-b33c-1a7bdc344b0b.mov

Great job as always!

File: packages/page-pattern-modal/src/pattern-definition.ts [hunk 3]
```diff
@@ -12,8 +14,16 @@ export interface PatternDefinition {
...
+	pattern_meta?: { is_mobile: boolean; is_web: boolean; viewport_width: number };
 	site_id?: number;
 	source_url?: string;
 	tags?: Record< string, { slug: string; title: string; description: string } >;
 }
+
+export interface FormattedPattern {
+	name: string;
+	title: string;
+	description: string;
```
Comment (fushar) [inline packages/page-pattern-modal/src/pattern-definition.ts]: Do we ever use this `description` prop anywhere? Maybe we can remove it.

**Review by fushar — APPROVED** [review]
It works really well! Code looks good too. But I have the same comments as @miksansegundo about the magic widths 😄

File: packages/page-pattern-modal/src/pattern-definition.ts (hunk 3 above)
Comment (fushar) [reply packages/page-pattern-modal/src/pattern-definition.ts]: Or better, use this props and remove `title` 😄

**Comment by arthur791004** [comment]
@miksansegundo @fushar Those default viewport widths are from [here](https://github.com/Automattic/wp-calypso/blob/d22976d8250fb4479d5677f5434742878fbd0ef3/apps/editing-toolkit/editing-toolkit-plugin/block-patterns/class-block-patterns-from-api.php#L115) to keep it consistent with the patterns from WP.com

File: packages/page-pattern-modal/src/pattern-definition.ts (hunk 3 above)
Comment (arthur791004) [reply packages/page-pattern-modal/src/pattern-definition.ts]: Removed the description. We have to keep `title` as it's used by [BlockPattern](https://github.com/WordPress/gutenberg/blob/6888113543212f5ceb29edd2cbfb184fece507e8/packages/block-editor/src/components/block-switcher/pattern-transformations-menu.js#L125)

File: packages/page-pattern-modal/src/components/page-pattern-modal.tsx (hunk 1 above)
Comment (arthur791004) [reply packages/page-pattern-modal/src/components/page-pattern-modal.tsx:53]: Added, thanks!

**Review by fushar — APPROVED** [review]

**Comment by miksansegundo** [comment]
Have you reproduced the issue that I found? How are the Link in bio previews looking for you?

**Comment by fushar** [comment]
> Have you reproduced the issue that I found? How are the Link in bio previews looking for you?

@miksansegundo sorry I missed your comment. I tried actually creating a new page using that pattern. In the Site Editor, I tried resizing the browser window and the behavior matches your description and what we see in the pattern picker -- when the viewport is a bit wider, it will look like "landscape mode".

So maybe it is indeed the behavior of the pattern and not related to this PR?

**Comment by miksansegundo** [comment]
> > Have you reproduced the issue that I found? How are the Link in bio previews looking for you?
> 
> @miksansegundo sorry I missed your comment. I tried actually creating a new page using that pattern. In the Site Editor, I tried resizing the browser window and the behavior matches your description and what we see in the pattern picker -- when the viewport is a bit wider, it will look like "landscape mode".
> 
> So maybe it is indeed the behavior of the pattern and not related to this PR?

Thanks for confirming this. It doesn't look related to this PR.

**Comment by arthur791004** [comment]
Yeah, there is an issue related to the height of the pattern. See https://github.com/WordPress/gutenberg/pull/38175 and that's why Gutenberg needs to define the [maxHeight](https://github.com/WordPress/gutenberg/blob/a556b6932d0163f1f23886bd003ad6c819cb4dc0/packages/block-editor/src/components/block-preview/auto.js#L108) of the pattern.

**Pull request merged**